src/Base/Parameter.cpp

```diff
@@ -361,14 +489,189 @@ bool ParameterGrp::HasGroup(const char* Name) const
     if (_GroupMap.find(Name) != _GroupMap.end())
         return true;
 
-    if (FindElement(_pGroupNode,"FCParamGroup",Name))
+    if (_pGroupNode && FindElement(_pGroupNode,"FCParamGroup",Name) != 0)
         return true;
 
     return false;
 }
 
+const char *ParameterGrp::TypeName(ParamType Type)
+{
+    switch(Type) {
+    case FCBool:
+        return "FCBool";
+    case FCInt:
+        return "FCInt";
+    case FCUInt:
+        return "FCUInt";
+    case FCText:
+        return "FCText";
+    case FCFloat:
+        return "FCFloat";
+    case FCGroup:
+        return "FCParamGroup";
+    default:
+        return 0;

```

Comment:
I think some static analyzers are going to flag this: consider `nullptr` (or throw something).